### PR TITLE
Remove obsolete references to starvation.

### PR DESCRIPTION
**Pull request by wheals**
Use generic "depopulated" instead.

It makes more sense for species like the Trith who don't need food, and food isn't a game concept anymore anyway.

Probably not worth cherry-picking since it changes the stringtable.

### CODE REVIEW

File: util/SitRepEntry.h [hunk 1]
```diff
@@ -54,7 +54,7 @@ FO_COMMON_API SitRepEntry CreateGroundCombatSitRep(int planet_id, int empire_id)
 FO_COMMON_API SitRepEntry CreatePlanetCapturedSitRep(int planet_id, int empire_id);
 FO_COMMON_API SitRepEntry CreateCombatDamagedObjectSitRep(int object_id, int combat_system_id, int empire_id);
 FO_COMMON_API SitRepEntry CreateCombatDestroyedObjectSitRep(int object_id, int combat_system_id, int empire_id);
-SitRepEntry               CreatePlanetStarvedToDeathSitRep(int planet_id);
+SitRepEntry               CreatePlanetDepopulatedSitRep(int planet_id);
```
Comment (geoffthemedio) [inline util/SitRepEntry.h:57]: I had to add FO_COMMON_API to get this to compile. Not sure why.

File: util/SitRepEntry.cpp [hunk 2]
```diff
@@ -246,7 +246,7 @@ SitRepEntry CreateCombatDestroyedObjectSitRep(int object_id, int combat_system_i
 }
 
 SitRepEntry CreatePlanetStarvedToDeathSitRep(int planet_id) {
```
Comment (geoffthemedio) [inline util/SitRepEntry.cpp]: Possibly because this function isn't renamed, but references to it are?

**Comment by wheals** [comment]
Ack! I guess the renaming got eaten in the rebase.

**Comment by wheals** [comment]
It now builds for me.

**Comment by geoffthemedio** [comment]
Builds OK and seems to run (and produce relevant sitreps) OK.

**Pull request merged**